### DISCUSSION TITLE
No Way for Filebeat ?

### DISCUSSION THREAD

**Discussion by wandelfalke** [Security-Onion-Solutions/securityonion#5565] [Unsupported Versions] [answered]
Hellouw

this will be my first post. so please ..  ^_^
Aaand im still new to Security Onion but,

It´s running. Yeeee  :)
All on the same machine ( Win 10 ) with 1 Network Card --> Security Onion ( v 2.3.70 ) is running in Hyper-V
After some lightyears of research ... got it running with mirroring in Hyper-v....
Running Winlogbeats on Windows ? no problem
Running Packetbeat on Windows ? no problem
Installing Wazuh Client on my Windows ? Yea when you learn to use the correct version!! ? no problem

But

i have a Suricata ( yes ) also running on my Windows machine. no problem
Sending the Log to SO2 ? ahh here we are
Using **Filebeat** on Windows and it wont work. 
After researching for another lightyear .. puh all goes wrong what ever i do.
Error message all the same again and again:
_ERROR	[publisher_pipeline_output]	pipeline/output.go:154	Failed to connect to backoff(async(tcp://x.x.x.x:5044)): dial tcp x.x.x.x:5044: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond._
( not realy x.x.x.x i hide the IPs )

I stoped the other beats .. still not working
using Rsyslog ? no way

So please whats wrong with this FILEBEAT ?

Thank you
Holger

**Comment by hacker0ni** [comment]
Make sure you use the latest version of Filebeat. Allow the filebeat agent from the firewall using `so-allow` (-b option, port 5044). Use Filebeat output to logstash and point to Security Onion. I can help you with further questions, but I'm also having problems with proper ingestion of logs.

**Reply by wandelfalke** [reply]
Hellouw
Sure. Winlog and Packetbeat working perfect on 5044!
Same fore Filebeat? No.
Not sure if Filebeat or maybe the Log (eve.json) self are the troublemaker..
Ah. Have to check the version but think was latest

**Reply by dougburks** [reply]
To clarify, you may not actually want the latest version of Filebeat. You want the version of Filebeat that matches the version of the Elastic stack in your Security Onion installation:
https://docs.securityonion.net/en/2.3/beats.html#version

**Reply by wandelfalke** [reply]
See my next post

**Reply by wandelfalke** [reply]
Version 7.13.4 worked for me so far ....

**Reply by dougburks** [reply]
Yes, if you're using our latest Security Onion 2.3.70 then it includes Elastic 7.13.4 and so Filebeat 7.13.4 is the correct version to use.

**Reply by wandelfalke** [reply]
Yes, Thank you
But with some Errors ..

**Comment by wandelfalke** [comment]
Soo Good Morning 

I dont know what happened ...
After this Weekend. It looks like he is working now.

_2021-09-20T13:04:09.557+0200	DEBUG	[logstash]	logstash/async.go:172	6 events out of 6 events sent to logstash host 10.1.3.216:5044. Continue sending_

I found things under "Security Onion" - Logs (?)
Dont find an Entry for Filebeat in "Host" Section in Elastic (?)

Aaand it looks like Elastic ( he is working with the Data ?) have some problems ..
Lets take a closer look:


_message | mapping update rejected by primary java.lang.IllegalArgumentException: Limit of total fields [1500] has been exceeded 	at org.elasticsearch.index.mapper.MappingLookup.checkFieldLimit(MappingLookup.java:216) ~[elasticsearch-7.13.4.jar:7.13.4] 	at_ 
etc etc etc

And:

_Multi fields  | message.keyword: mapping update rejected by primary java.lang.IllegalArgumentException: Limit of total fields [1500] has been exceeded 	at org.elasticsearch.index.mapper.MappingLookup.checkFieldLimit(MappingLookup.java:216) ~[elasticsearch-7.13.4.jar:7.13.4] 	at org.elasticsearch.index.mapper.MappingLookup.checkLimits(MappingLookup.java:208) ~[elasticsearch-7.13.4.jar:7.13.4]_


Thanks :)

**Comment by wandelfalke** [comment]
So, i reached the end of the internet .. again
for my understanding.
filebeat sends the eve.json to Logstash.
check that works.
But, Logstash ( within security onion 2 ) does nothing with it and sends it to Elasticsearch. ?
and i think the error within Elastic:
mapping update rejected by primary
java.lang.IllegalArgumentException: Limit of total fields [1500] has been exceeded

comes from Elastic, the Pipeline cant handle the Log ??
what next ?

Thank you

**Comment by dougburks** [comment, marked as answer]
Looking back at your original post, it seems like you are trying to get Suricata logs from your Windows host to Security Onion. However, Security Onion already has Suricata built in so if it's monitoring your host traffic, then there is no need to run Suricata on your host.

**Reply by wandelfalke** [reply]
Yes this is totaly true :)
but i think i dont get the real traffic in security onion ...
Hyper-v with 1 vSwitch as a sender ( mirror ) and 2nd vSwitch ( for SO ) as receiver ..
looks like it doesnt work this way.
so i go away from this setting.
i place suricata on my windows directly and analyze the traffic direct.

**Reply by dougburks** [reply]
It should be possible to configure Hyper-V for proper sniffing as folks have reported this working in the past. This would be the better method since it would allow you to take advantage of not only Suricata, but Zeek and full packet capture on Security Onion as well.
